modules/apps/roles/roles-test/src/testIntegration/java/com/liferay/roles/service/test/RoleLocalServiceSystemRolesTest.java

```diff
@@ -112,15 +120,62 @@ public void testAnalyticsAdminRolePermissions() throws Exception {
 		}
 	}
 
+	@Test
+	public void testCheckSystemnRole() throws Exception {
+		long companyId = TestPropsValues.getCompanyId();
+		long groupId = TestPropsValues.getGroupId();
+		_user = UserTestUtil.addUser();
+
+		Role powerUserRole = _roleLocalService.fetchRole(
+			TestPropsValues.getCompanyId(), RoleConstants.POWER_USER);
+
+		Role userRole = _roleLocalService.fetchRole(
+			TestPropsValues.getCompanyId(), RoleConstants.USER);
+
+		PermissionChecker permissionChecker =
+			PermissionCheckerFactoryUtil.create(_user);
+
+		boolean hasViewPermission = permissionChecker.hasPermission(
+			groupId, Role.class.getName(), powerUserRole.getRoleId(),
+			ActionKeys.VIEW);
+
+		_resourcePermissionLocalService.removeResourcePermission(
+			companyId, Role.class.getName(), ResourceConstants.SCOPE_INDIVIDUAL,
+			String.valueOf(powerUserRole.getRoleId()), userRole.getRoleId(),
+			ActionKeys.VIEW);
+
+		_roleLocalService.checkSystemRoles(companyId);
+
+		boolean testViewPermission = permissionChecker.hasPermission(
+			groupId, Role.class.getName(), powerUserRole.getRoleId(),
+			ActionKeys.VIEW);
+
+		Assert.assertFalse(testViewPermission);
+		
+		if (hasViewPermission) {
+			_resourcePermissionLocalService.setResourcePermissions(
+				companyId, Role.class.getName(),
+				ResourceConstants.SCOPE_INDIVIDUAL, 
+				String.valueOf(powerUserRole.getRoleId()), userRole.getRoleId(),
+				new String[] {ActionKeys.VIEW});
+		}
```

Comment:
@hudakl This should be inside of a `try/finally` block so that it will run even if an exception is thrown during the test case. 142-153 should be inside `try`, then 155-161 inside the `finally`.